src/containers/SigninForm/signInHandler.ts

```diff
@@ -0,0 +1,94 @@
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+
+import Pages from '@/constants/pages';
+import authService from '@/services/authService';
+import { AuthCredentials, IApiError, IUserLoginResponseMessage } from '@/constants/types';
+
+interface ResponseMessage {
+  status: 'success' | 'error' | '';
+  title: string;
+  message: string;
+}
+
+const ERROR_MESSAGES = {
+  USER_NOT_EXIST: 'User with this Email not found, Sign up instead.',
+  INVALID_CREDENTIALS: 'Email or password does not match with your information in our database.',
+  AUTH_FAILED: 'An error occurred during authentication, try again.',
+};
+
+const SUCCESS_MESSAGE: ResponseMessage = {
+  status: 'success',
+  title: 'Login successful',
+  message: 'Your login was successful, now we will redirect you to the dashboard.',
+};
+
+const SignInHandler = (
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean>>,
+  isRememberChecked: boolean,
+) => {
+  const router = useRouter();
+  const [response, setResponse] = useState<ResponseMessage>({
+    status: '',
+    title: '',
+    message: '',
+  });
+
+  const handleCloseModal = () => {
+    setIsOpen(false);
+  };
+
+  const handleAuthSuccess = (token: string) => {
+    setResponse(SUCCESS_MESSAGE);
+    if (isRememberChecked) {
+      localStorage.setItem('token', token);
+    } else {
+      sessionStorage.setItem('token', token);
+    }
```

Comment:
instead of doing this, save the `isRememberChecked` somewhere (in localStorage and then redux when the app loads preferably) and then, create another function called storage, which checks this `isRemember` value in redux/localStroage, if the value is false, it returns `sessionStorage`. if the value is true, it returns `localStorage`.

Then, you can just run = `storage().setItem('token', token)` or something like this.